test/integration_test.py

```diff
@@ -707,8 +707,12 @@ def test_concurrent_portal_db_crud(self):
 
         entry_format = 'task={};op={}'
 
+        running = True
+
         def run(thread_count):
             for op_count in range(n_ops):
+                if not running:
+                    return
                 mock_entry = {
```

Comment:
Please check if the shut down sequence still works if you put a time.sleep(1) before this line. I'm trying to force a race condition that way.

Reply:
For both the happy and the unhappy path, that is.